src/app/pages/sharing/smb/smb-form/smb-form.component.spec.ts

```diff
@@ -0,0 +1,449 @@
+    const optionLabels = await purposeSelect.getOptionLabels();
+    expect(optionLabels).toEqual([
+      '--',
+      'No presets',
+      'Default share parameters',
+      'Multi-user time machine',
+      'Multi-protocol (NFSv3/SMB) shares',
+      'Private SMB Datasets and Shares',
+      'SMB WORM. Files become readonly via SMB after 5 minutes',
+    ]);
+  });
+
+  it('should have error for duplicate share name', async () => {
+    const nameControl = await loader.getHarness(IxInputHarness.with({ label: 'Name' }));
+    await nameControl.setValue('ds222');
+    expect(await nameControl.getErrorText()).toEqual('The name "ds222" is already in use.');
+  });
+
+  it('when a preset is selected, the relevent fields should be impacted', async () => {
+    const advancedButton = await loader.getHarness(MatButtonHarness.with({ text: 'Advanced Options' }));
+    await advancedButton.click();
+
+    const purposeSelect = await loader.getHarness(IxSelectHarness.with({ label: formLabels.purpose }));
+
+    const labels = await purposeSelect.getOptionLabels(); /** 1 so to Skip '--' */
+    labels.splice(0, 1);
+    const presetKeys = Object.keys(presets);
+    const form = await loader.getHarness(IxFormHarness);
+    const fields = await form.getControlHarnessesDict();
+
+    for (let i = 0; i < labels.length; i++) {
+      await purposeSelect.setValue(labels[i]);
+      for (const param in presets[presetKeys[i]].params) {
+        if (param === 'auxsmbconf') {
+          continue;
+        }
+        const expectedValue = presets[presetKeys[i]].params[param as keyof SmbShare];
+        const value = await fields[formLabels[param]].getValue();
+        expect(value).toStrictEqual(expectedValue);
+        expect(await fields[formLabels[param]].isDisabled()).toBeTruthy();
+      }
+    }
+    expect(true).toBeTruthy();
+  });
+
+  it('should show confirmation warning when afp is checked', async () => {
+    const advancedButton = await loader.getHarness(MatButtonHarness.with({ text: 'Advanced Options' }));
+    await advancedButton.click();
+    const afpCheckbox = await loader.getHarness(IxCheckboxHarness.with({ label: formLabels.afp }));
+    await afpCheckbox.setValue(true);
+    expect(spectator.inject(DialogService).confirm).toHaveBeenNthCalledWith(1, {
+      title: helptextSharingSmb.afpDialog_title,
+      message: helptextSharingSmb.afpDialog_message,
+      hideCheckBox: false,
+      buttonMsg: helptextSharingSmb.afpDialog_button,
+      hideCancel: false,
+    });
+  });
+
+  it('shows values of existing share when editing', async () => {
+    spectator.component.setSmbShareForEdit(existingShare);
+
+    const advancedButton = await loader.getHarness(MatButtonHarness.with({ text: 'Advanced Options' }));
+    await advancedButton.click();
+
+    const values = await form.getValues();
+
+    const existingShareWithLabels: { [key: string]: any } = {};
+    for (const key in existingShare) {
```

Comment:
This is my personal opinion, so no change necessary for this PR – I find it easier to read when tests are less programmatic and more close to what user would see, even if it's more lines, i.e. just showing labels and their values.